### PR TITLE
Helvetica regular toggle update

### PR DESCRIPTION
**Pull request by davidpmccormick**
## Who is this for?
People looking at using roman/bold weight of Helvetica Neue behind a toggle

## What is it doing for them?
Changing the way link styling works (using a combination of `text-decoration-underline`, `text-underline-offset` and `text-decoration-color`) to prevent larger gaps that appear as a result of using `border-bottom` on the webfont when Helvetica isn't available as a system font.

Also allowing the toggles to be accessible within the `<GlobalStyle />`

The webfont has some extra space above each character, causing text in buttons and alongside icons to appear too low down. I've added an `AlignFont` component to apply an eye-balled amount of negative `translateY` to compensate. I think the ideal solution would be to apply the required [ascent-override](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/ascent-override) at the `@font-face` level, but this isn't yet supported in Safari/mobile Safari. Keeping an eye on [this webkit bug](https://bugs.webkit.org/show_bug.cgi?id=219735) for implementation though.

__Example body anchor font weight/color__
image

__TODO__
Updating `letter-spacing` for the `roman` weight of Helvetica Neue to be fractionally more spaced out. Unfortunately this isn't possible on the `@font-face` declaration, so not possible without targeting classes.

### CODE REVIEW

**Review by jamesgorrie — APPROVED** [review]

File: common/views/components/styled/AlignFont.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  children: ReactNode;
+};
+
+const Align = styled.span<{ isOn: boolean }>`
+  .fonts-loaded & {
+    ${props =>
+      props.isOn &&
+      `
+      transform: translateY(-${props.theme.fontVerticalOffset});
+      display: inline-flex;
```
Comment (gestchild) [inline common/views/components/styled/AlignFont.tsx:16]: why do we need inline-flex?

File: common/views/components/styled/AlignFont.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const AlignFont: FunctionComponent<Props> = ({ children, className }) => {
+  const { helveticaRegular } = useContext(TogglesContext);
+  return (
+    <Align isOn={helveticaRegular} className={className}>
+      {children}
+    </Align>
+  );
+};
+
+export default AlignFont;
```
Comment (gestchild) [inline common/views/components/styled/AlignFont.tsx:30]: Could we add a comment about when we should use this and why it's needed?

File: common/views/components/styled/AlignFont.tsx (hunk 1 above)
Comment (davidpmccormick) [reply common/views/components/styled/AlignFont.tsx:16]: It has to be something other than `inline` for the `translateY` to have any effect. Could be `inline-block`, maybe?

File: common/views/components/styled/AlignFont.tsx (hunk 2 above)
Comment (davidpmccormick) [reply common/views/components/styled/AlignFont.tsx:30]: yeah, good point

**Review by gestchild — APPROVED** [review]

**Pull request merged**